csharp/src/Ice/IceSSL/TransceiverI.cs

```diff
@@ -360,93 +358,57 @@ private bool startAuthenticate(IceInternal.AsyncCallback callback, object state)
+                        new RemoteCertificateValidationCallback(validationCallback)));
+                _writeResult.ContinueWith(
+                    task =>
+                    {
+                        try
                         {
-                            try
-                            {
-                                // If authentication fails, AuthenticateAsClientAsync throws AuthenticationException,
-                                // and the task won't complete successfully.
-                                _verified = task.IsCompletedSuccessfully;
-                                if (_verified)
-                                {
-                                    _cipher = _sslStream.CipherAlgorithm.ToString();
-                                    if (_sslStream.RemoteCertificate is X509Certificate2 remoteCertificate)
-                                    {
-                                        _certs = [remoteCertificate];
-                                    }
-                                }
-                            }
-                            finally
+                            // If authentication fails, AuthenticateAsServerAsync throws AuthenticationException,
+                            // and the task won't complete successfully.
+                            _verified = task.IsCompletedSuccessfully;
+                            if (_verified)
                             {
-                                callback(state);
+                                _cipher = _sslStream.CipherAlgorithm.ToString();
                             }
-                        },
-                        TaskScheduler.Default);
-                }
-                else
-                {
-
-                    _writeResult = _sslStream.AuthenticateAsClientAsync(
-                        _host,
-                        _instance.certs(),
-                        _instance.checkCRL() > 0);
-                    _writeResult.ContinueWith(task => callback(state), TaskScheduler.Default);
-                }
+                        }
+                        finally
+                        {
+                            callback(state);
+                        }
+                    },
+                    TaskScheduler.Default);
             }
             else
             {
-                // Server authentication.
-                if (_serverAuthenticationOptions is not null)
-                {
-                    _writeResult = _sslStream.AuthenticateAsServerAsync(_serverAuthenticationOptions);
-                    _writeResult.ContinueWith(
-                        task =>
+                _writeResult = _sslStream.AuthenticateAsClientAsync(
+                    _instance.initializationData().clientAuthenticationOptions ??
+                    _instance.engine().clientAuthenticationOptions(
+                        new RemoteCertificateValidationCallback(validationCallback),
```

Comment:
Is this `new RemoteCertificateValidationCallback` necessary? I would think you can pass `validationCallback` directly (the delegate captures 'this').

Reply:
Fixed as suggested.